Update profiles specification README.md and pprof.md (#4932)
open-telemetry/opentelemetry-specification: specification/profiles/README.md

# OpenTelemetry Profiles

**Status**: [Development](../document-status.md)

> [!NOTE]
>
> The profiles signal is still experimental and under active development.

> > Breaking changes may be introduced in future versions.

Author: I removed this when I updated status from Development to Alpha.

# OpenTelemetry Profiles

**Status**: [Development](../document-status.md)

> [!NOTE]
>

> > The profiles signal is still experimental and under active development.

Reviewer: We should rephrase this note to sound less concerning. 
```suggestion
> The profiles signal is currently marked as Alpha under development.
```

Reviewer 2: Agreed. It's not yet marked as alpha (see [this PR](https://github.com/open-telemetry/opentelemetry-proto/pull/775)).

But maybe we don't need this whole paragraph since the Status is already declared above. Other documents with status "Development" don't seem to have such a Notes section either.

Anyway, since it's already in the other doc, I'm also fine with keeping it for now.

Reviewer 3: We should probably gather ourselves a list of all places that need to be coherently updated on the Alpha day.

  and in most cases supports lossless conversions to and from these formats.
  If that's not possible (e.g. custom extensions), the `original_payload_format`
  field can be used to transmit the original information for future lossless
  export or reinterpretation.
- **Correlation with other signals**: Profiles MUST be linkable to logs, metrics
  and traces through shared resource context and, where applicable,
  direct trace/span references. This is the primary differentiator of OpenTelemetry
  profiles compared to standalone profiling solutions.

## Data model

The OpenTelemetry profiles data model is defined [here](./data-model.md).
It builds on the [pprof protobuf format](https://github.com/google/pprof/tree/main/proto)
and extends it with:

- **Resource and scope context**: Each batch of profiles is associated with a
  [Resource](../resource/README.md) and an [InstrumentationScope](../common/instrumentation-scope.md),
  consistent with logs, metrics and traces.
- **Generalized dictionary**: Deduplicates not only strings but also other messages
  that exhibit redundancy.
- **Generalized attributes**: Most messages can carry attributes following the

>   same conventions as other signals, augmented with Unit information (`KeyValueAndUnit`).

Reviewer: As the topic around KeyValueAndUnit is under discussion and there will be some change, we should not point it out here. To me, its an implementation detail rather than a key concept in the data model.
Also, I think it does not add more to the data model that is not yet covered by `**Generalized dictionary**`.
```suggestion
```

Author: I think we absolutely need to mention this here for two main reasons:

1. This is a major differentiator from pprof
2. It doesn't follow existing OTel conventions so questions may arise 

I think one job of the spec is to clarify such corner cases instead of hiding them. Also, this is not an implementation detail but a core aspect of the data model, which I'll also try to explain in `data-model.md`.
